.github/workflows/test.yml

```diff
@@ -1,34 +1,73 @@
 name: test
 
-on: workflow_dispatch
-
-env:
-  FOUNDRY_PROFILE: ci
+on:
+  push:
+      branches:
+          - master
+  pull_request:
 
 jobs:
-  check:
-    strategy:
-      fail-fast: true
-
-    name: Foundry project
+  build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-        with:
-          submodules: recursive
-
-      - name: Install Foundry
-        uses: foundry-rs/foundry-toolchain@v1
-        with:
-          version: nightly
-
-      - name: Run Forge build
-        run: |
-          forge --version
-          forge build --sizes
-        id: build
-
-      - name: Run Forge tests
-        run: |
-          forge test -vvv
-        id: test
+        - uses: actions/checkout@v3
+          with:
+              submodules: recursive
+
+        - name: Install foundry
+          uses: foundry-rs/foundry-toolchain@v1
+          with:
+              version: nightly
+
+        - name: Run foundry build
+          run: |
+              forge --version
+              forge build
+          id: build
+
+  lint-check:
+      runs-on: ubuntu-latest
+      needs: build
+      steps:
+          - uses: actions/checkout@v3
+            with:
+                submodules: recursive
+          - uses: foundry-rs/foundry-toolchain@v1
+            with:
+                version: nightly
+
+          - name: Run foundry fmt check
+            run: |
+                forge fmt --check
+            id: fmt
+
+  test:
+      runs-on: ubuntu-latest
+      needs: lint-check
+      steps:
+          - uses: actions/checkout@v3
+            with:
+                submodules: recursive
+          - uses: foundry-rs/foundry-toolchain@v1
+            with:
+                version: nightly
+          - name: Run foundry tests
+            # --ast tests enables inline configs to work https://github.com/foundry-rs/foundry/issues/7310#issuecomment-1978088200
+            run: |
+                forge test -vvv --gas-report --ast 
+            id: test
+
+  coverage:
```

Comment:
I was wondering about this in the other repo as well, why do we run coverage in the CI?

Reply:
Haythem set it up this way for the EVK and to be honest I simply copy-pasted the file here. I don't think it's an issue though